### PR TITLE
Ped sync improvements

### PR DESCRIPTION
**Pull request by tederis**
This PR is a reimplementation of #800 with additional fixes in rotation interpolation. 

The video below demonstrates that even without animations a ped is well synchronized (in comparison to the pre-PR). 
https://youtu.be/17vEh7k_RSI?si=8kxGwODs7lIn8OaF

Note that since peds are correctly syncronized now, there may be problems with old scripts relying on a ped's animation speed. These scripts will have to use animation without a root motion or use `setPedAnalogControlState`.

### CODE REVIEW

**Comment by Fernando-A-Rocha** [comment]
Do you have a video of how it looks without this PR? 😁

**Comment by tederis** [comment]
> Do you have a video of how it looks without this PR? 😁

There's the old video (from the previous PR) with the pre-PR behaviour: https://youtu.be/UcSOXd9PxhE?si=cbNeF8FVg-xf5tBa

**Comment by Fernando-A-Rocha** [comment]
Nice, good job. The difference is notable.

**Review by TheNormalnij — DISMISSED** [review]

**Comment by TheNormalnij** [comment]
lgtm

**Review by Lpsd — APPROVED** [review]

File: Shared/sdk/CVector.h [hunk 1]
```diff
@@ -26,11 +26,15 @@ class CVector
...
-    float fY = 0.0f;
-    float fZ = 0.0f;
+    float fX;
+    float fY;
+    float fZ;
 
-    constexpr CVector() = default;
+    struct NoInit{};
+
+    CVector(NoInit) {}
```
Comment (TracerDS) [inline Shared/sdk/CVector.h:35]: Whats the point behind it?

Comment (tederis) [reply Shared/sdk/CVector.h:35]: This is the call to a special constructor without vector's components initialization. For the sake of a performance.

Comment (tederis) [reply Shared/sdk/CVector.h:35]: See for example https://github.com/rheit/zdbsp/blob/master/tarray.h or https://docs.unrealengine.com/4.27/en-US/API/Runtime/Core/UObject/FNameEntryId/

**Review by botder — APPROVED** [review]

**Pull request merged**